test/Constraints/invalid_decl_ref.swift

```diff
@@ -0,0 +1,9 @@
+// RUN: %target-swift-frontend -module-name SomeModule -typecheck -verify -dump-ast -import-objc-header %S/Inputs/imported_type.h %s | %FileCheck %s
+
+// REQUIRES: objc_interop
+
+import Foundation
+
+// CHECK: declref_expr type='module<SomeModule>'
+// CHECK-NEXT: type_expr type='Data.Type'
+let type = SomeModule.Data.self
```

Comment:
@brentdax You might be interested in this example because, from how @hborla described this to me it appears that `SomeModule.Data` lookup would produce multiple results here - one for `SomeModule.Data` and one for `Foundation.Data` which seems like a bug in shadowing resolution?